### PR TITLE
mailling api communication

### PR DESCRIPTION
**Pull request by maxime-dodin**
#### Description

Allow communication from notes-service to accounts-service,
SendInvite call SendGroupInviteMail in accounts-service to notify the recipient by mail

Fixes #<issue number>

#### Changelog

- [BUGFIX]
- [ENHANCEMENT]
- [DOCS]
- ...

### CODE REVIEW

**Review by edouard-sn — APPROVED** [review]
Would have done the same 😎

File: invites.go [hunk 1]
```diff
@@ -38,12 +38,23 @@ func (srv *groupsAPI) SendInvite(ctx context.Context, req *notesv1.SendInviteReq
...
 		}
+		inviteGroup, err := srv.groups.GenerateGroupInviteLink(ctx,
+			&models.OneGroupFilter{GroupID: req.GroupId},
+			&models.GenerateGroupInviteLinkPayload{
+				GeneratedByAccountID: invite.RecipientAccountID,
+				ValidUntil:           invite.ValidUntil,
+			}, token.AccountID)
+		if err != nil {
+			return nil, statusFromModelError(err)
+		}
```
Comment (edouard-sn) [inline invites.go:49]: InviteLink is not the same as Invite!
We do not generate an InviteLink in SendInvite

File: invites.go [hunk 2]
```diff
@@ -47,12 +47,14 @@ func (srv *groupsAPI) SendInvite(ctx context.Context, req *notesv1.SendInviteReq
 		if err != nil {
 			return nil, statusFromModelError(err)
 		}
+
+		inviteLink := "https://noted-eip.vercel.app/groups/" + req.GroupId + "/invite/" + inviteGroup.Code
```
Comment (edouard-sn) [inline invites.go:51]: I'm gonna explain Discord

**Review by edouard-sn — CHANGES_REQUESTED** [review]

File: invites.go (hunk 1 above)
Comment (b4chh) [reply invites.go]: I have fixed the code

File: invites.go (hunk 2 above)
Comment (b4chh) [reply invites.go]: Thank you for your explication

File: communication/accounts.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import (
+	accountsv1 "notes-service/protorepo/noted/accounts/v1"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type AccountsServiceClient struct {
+	conn     *grpc.ClientConn
```
Comment (gabriel654165) [inline communication/accounts.go:11]: Why do you store the `grpc.ClientConn` in conn ?
The connection is no longer usefull after the newAPIClient created.

File: server.go [hunk 4]
```diff
@@ -32,6 +33,7 @@ type server struct {
 	authService       auth.Service
 	backgroundService background.Service
 	languageService   language.Service // NOTE: Could put directly service typed as NaturalAPIService, remove Init() from interface and just put it in NaturalAPIService
+	accountService    *communication.AccountsServiceClient
```
Comment (gabriel654165) [inline server.go:36]: Why creating a service for connection and don't you just store a `accountsv1.AccountsAPIClient` in the server ?

**Review by gabriel654165 — CHANGES_REQUESTED** [review]

File: server.go (hunk 4 above)
Comment (maxime-dodin) [reply server.go:36]: in my opinion it's cleaner to wrap the connection with other services in another package

File: communication/accounts.go (hunk 3 above)
Comment (maxime-dodin) [reply communication/accounts.go:11]: make it easier to close the connection at the end

**Pull request closed without merging**